src/callback/file_callbacks.cpp

```diff
@@ -98,12 +98,11 @@ namespace Callbacks{
         fileThread = std::thread(&GodotFileRunner::run, this);
     }
 
-    void GodotFileRunner::finish(){
+    void GodotFileRunner::finish() {
         stop = true;
         //we need to notify the loop one last time, otherwise it will stay stuck in the wait method.
         read_cv.notify_one();
-        fileThread.join();
-
+        //fileThread.join();
```

Comment:
For some reason this crashes Godot 4 when trying to exit it. I'll comment it for now but we should look into it.